### PR TITLE
Handle http parser errors

### PR DESCRIPTION
**Pull request by millicentachieng**

### CODE REVIEW

File: ApiDoctor.Validation/Http/HttpParser.cs [hunk 1]
```diff
@@ -66,9 +67,9 @@ public HttpRequest ParseHttpRequest(string requestString)
                         var components = line.Split(' ');
 
                         if (components.Length < 2)
-                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line.");
+                            throw new HttpParserRequestException($"Request does not contain a proper HTTP request first line. Allowed format examples: GET https://graph.microsoft.com/beta/accessReviews(id) or GET HTTP/1.1  https://graph.microsoft.com/beta/accessReviews(id)");
```
[the remark this answers is not in the document]
Comment (andrueastman) [reply ApiDoctor.Validation/Http/HttpParser.cs:70]: https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html

File: ApiDoctor.Validation/MethodDefinition.cs [hunk 2]
```diff
@@ -468,24 +481,17 @@ internal static void RewriteHeadersWithParameters(HttpRequest request, IEnumerab
         /// </summary>
         internal void VerifyRequestFormat(IssueLogger issues)
         {
-            HttpParser parser = new HttpParser();
             HttpRequest request;
-            try
+            HttpParser.TryParseHttpRequest(this.Request, out request);
```
Comment (andrueastman) [inline ApiDoctor.Validation/MethodDefinition.cs:485]: Looks like you want to pass in the issues here as well when you call `TryParseHttpRequest`

File: ApiDoctor.Validation/DocFile.cs [hunk 3]
```diff
@@ -1211,11 +1236,12 @@ public List<ItemDefinition> ParseCodeBlock(Block metadata, Block code, IssueLogg
                                 MethodDefinition pairedRequest = (from m in this.requests where m.Identifier == requestMethodName select m).FirstOrDefault();
                                 if (pairedRequest != null)
                                 {
-                                    pairedRequest.AddExpectedResponse(code.Content, annotation);
+                                    pairedRequest.AddExpectedResponse(GetBlockContent(code), annotation);
                                     responses.Add(pairedRequest);
                                 }
                                 else
                                 {
+                                    //issues.Error(ValidationErrorCode.MarkdownParserError, $"Unable to locate the corresponding request for response block: {annotation.MethodName}. Requests must be defined before a response.");
```
Comment (andrueastman) [inline ApiDoctor.Validation/DocFile.cs:1244]: [NIT] Commented out code added

File: ApiDoctor.Validation/DocFile.cs [hunk 4]
```diff
@@ -1228,18 +1254,19 @@ public List<ItemDefinition> ParseCodeBlock(Block metadata, Block code, IssueLogg
...
                                 try
                                 {
-                                    pairedRequest.AddExpectedResponse(code.Content, annotation);
+                                    pairedRequest.AddExpectedResponse(GetBlockContent(code), annotation);
                                     responses.Add(pairedRequest);
                                 }
                                 catch (Exception ex)
                                 {
                                     detectedErrors.Add(new ValidationError(ValidationErrorCode.MarkdownParserError, this.DisplayName, "Unable to pair response with request {0}: {1}.", annotation.MethodName, ex.Message));
+                                    //issues.Error(ValidationErrorCode.MarkdownParserError, $"Unable to pair response with request {annotation.MethodName}", ex);
```
Comment (andrueastman) [inline ApiDoctor.Validation/DocFile.cs:1263]: [NIT] Commented out code added

File: ApiDoctor.Validation/methodvalidationextensionmethods.cs [hunk 5]
```diff
@@ -73,7 +73,7 @@ public static class MethodValidationExtensionMethods
                         RequiredTags = method.RequiredTags
                     }
                 };
-//                results.AddResult("init", new ValidationMessage(null, "No scenarios were defined for method {0}. Will request verbatim from docs.", method.Identifier), ValidationOutcome.None);
+                // results.AddResult("init", new ValidationMessage(null, "No scenarios were defined for method {0}. Will request verbatim from docs.", method.Identifier), ValidationOutcome.None);
```
Comment (andrueastman) [inline ApiDoctor.Validation/methodvalidationextensionmethods.cs:76]: [NIT] Commented code

File: ApiDoctor.Validation/Http/HttpParser.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (millicentachieng) [reply ApiDoctor.Validation/Http/HttpParser.cs:70]: Sure, thanks for pointing out this

File: ApiDoctor.Validation/MethodDefinition.cs (hunk 2 above)
Comment (millicentachieng) [reply ApiDoctor.Validation/MethodDefinition.cs:485]: At this point, validation of the HttpRequest had already happened and any issues logged. Adding issues here causes duplicated errors if any.

File: ApiDoctor.Console/Program.cs [hunk 6]
```diff
@@ -675,28 +675,20 @@ private static async Task<CheckResults> CheckMethodsAsync(BasicCheckOptions opti
 
                 TestReport.StartTest(testName, method.SourceFile.DisplayName, skipPrintingHeader: options.PrintFailuresOnly);
 
-                if (string.IsNullOrEmpty(method.ExpectedResponse))
+                if (method.ExpectedResponse == null)
```
[the remark this answers is not in the document]
Comment (millicentachieng) [reply ApiDoctor.Console/Program.cs:678]: To differentiate between an empty and non-existent response block

**Pull request merged**